Util/channel/src/main/java/io/deephaven/util/channel/LocalFSChannelProvider.java

```diff
@@ -56,6 +60,20 @@ public SeekableByteChannel getWriteChannel(@NotNull final Path filePath, final b
         return result;
     }
 
+    @Override
+    public final Stream<URI> list(@NotNull final URI directory) throws IOException {
+        // Assuming that the URI is a file, not a directory. The caller should manage file vs. directory handling in
+        // the processor.
+        return Files.list(Path.of(directory)).map(path -> FileUtils.convertToURI(path, false));
```

Comment:
`Files.list` wants a try-with-resources. Maybe you need a similar warning on this method, assuming `.close()` chains backwards. If not, you may need to produce a `List` and return a `Stream` over it.

Reply:
I told Shivam that Stream was better than List; close() does chain backwards, so as long as consumers use correctly, we should be good w/ javadoc update.

Reply:
Added the javadoc.